middleman-core/lib/middleman-core/builder.rb

```diff
@@ -124,7 +129,7 @@ def run!
       unless @has_error
         partial_update_with_no_changes = (@only_changed || @missing_and_changed) && @invalidated_files.empty?
 
-        ::Middleman::Dependencies.serialize_and_save(@app, @graph) if @track_dependencies && !partial_update_with_no_changes
+        ::Middleman::Dependencies.serialize_and_save(@app, @graph) if @track_dependencies && (create_deps_yml || !partial_update_with_no_changes)
```

Comment:
I bailed on adding specs here because `run!` is really long. I tested it manually and it works though.